qa/L0_perf_resnet/run_test.sh

```diff
@@ -43,95 +42,77 @@ export CUDA_VISIBLE_DEVICES=0
-            fi
-
-            SERVER_LOG="${NAME}.serverlog"
-            run_server
-            if (( $SERVER_PID == 0 )); then
-                echo -e "\n***\n*** Failed to start $SERVER\n***"
-                cat $SERVER_LOG
-                exit 1
-            fi
-
-            set +e
-
-            # Run the model once to warm up. Some frameworks do
-            # optimization on the first requests.
-            $PERF_CLIENT -v -i grpc -m $MODEL_NAME -p5000 -b${STATIC_BATCH}
-
-            $PERF_CLIENT -v -i grpc -m $MODEL_NAME -p5000 \
-                         -b${STATIC_BATCH} --concurrency-range ${CONCURRENCY} \
-                         -f ${NAME}.csv >> ${NAME}.log 2>&1
-            if (( $? != 0 )); then
-                RET=1
-            fi
-            curl localhost:8002/metrics -o ${NAME}.metrics >> ${NAME}.log 2>&1
-            if (( $? != 0 )); then
-                RET=1
-            fi
-
-            set -e
-
-            echo -e "[{\"s_benchmark_kind\":\"benchmark_perf\"," >> ${NAME}.tjson
-            echo -e "\"s_benchmark_name\":\"resnet50\"," >> ${NAME}.tjson
-            echo -e "\"s_protocol\":\"grpc\"," >> ${NAME}.tjson
-            echo -e "\"s_framework\":\"${MODEL_FRAMEWORK}\"," >> ${NAME}.tjson
-            echo -e "\"s_model\":\"${MODEL_NAME}\"," >> ${NAME}.tjson
-            echo -e "\"l_concurrency\":${CONCURRENCY}," >> ${NAME}.tjson
-            echo -e "\"l_dynamic_batch_size\":${DYNAMIC_BATCH}," >> ${NAME}.tjson
-            echo -e "\"l_batch_size\":${STATIC_BATCH}," >> ${NAME}.tjson
-            echo -e "\"l_instance_count\":${INSTANCE_CNT}}]" >> ${NAME}.tjson
-
-            kill $SERVER_PID
-            wait $SERVER_PID
-
-            if [ -f $REPORTER ]; then
-                set +e
-
-                URL_FLAG=
-                if [ ! -z ${BENCHMARK_REPORTER_URL} ]; then
-                    URL_FLAG="-u ${BENCHMARK_REPORTER_URL}"
-                fi
-
-                $REPORTER -v -o ${NAME}.json --csv ${NAME}.csv ${URL_FLAG} ${NAME}.tjson
-                if (( $? != 0 )); then
-                    RET=1
-                fi
-
-                set -e
-            fi
-        done
-    done
-done
+MAX_BATCH=${STATIC_BATCH}
+CONCURRENCY=$(( $INSTANCE_CNT * 2 ))
+if (( $CONCURRENCY < $REQUIRED_CONCURRENCY )); then
+    CONCURRENCY=$REQUIRED_CONCURRENCY
+fi
+
+NAME=${MODEL_NAME}_sbatch${STATIC_BATCH}_instance${INSTANCE_CNT}_${PERF_CLIENT_PROTOCOL}
+
+rm -fr models && mkdir -p models && \
+    cp -r $MODEL_PATH models/. && \
+    (cd models/$MODEL_NAME && \
+            sed -i "s/^max_batch_size:.*/max_batch_size: ${MAX_BATCH}/" config.pbtxt && \
+            echo "instance_group [ { count: ${INSTANCE_CNT} }]" >> config.pbtxt)
+
+SERVER_LOG="${NAME}.serverlog"
+run_server
+if (( $SERVER_PID == 0 )); then
+    echo -e "\n***\n*** Failed to start $SERVER\n***"
+    cat $SERVER_LOG
+    exit 1
+fi
+
+
+set +e
+
+PERF_CLIENT_PROTOCOL_ARGS="-i grpc -u localhost:8001" &&
```

Comment:
You can directly pass the -i  $PERF_CLIENT_PROTOCOL as the cli option, the perf_client will automaitcally select the appropriate url and port number.